src/test/java/sorald/processor/ProcessorDescriptionTest.java

```diff
@@ -0,0 +1,39 @@
+package sorald.processor;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FilenameUtils;
+import org.junit.jupiter.api.Test;
+import sorald.Processors;
+
+public class ProcessorDescriptionTest {
+    private static final Path PROCESSOR_PACKAGE = Paths.get("src/main/java/sorald/processor");
+
+    @Test
+    public void test_eachProcessorIsAccompaniedByDescription() {
+        List<File> processors =
+                Processors.getAllProcessors().stream()
+                        .map(Class::getSimpleName)
+                        .map(procName -> PROCESSOR_PACKAGE.resolve(procName + ".java"))
+                        .map(Path::toFile)
+                        .collect(Collectors.toList());
+        processors.forEach(
+                processor ->
+                        assertTrue(
+                                getDescription(processor).isFile(),
+                                "Description corresponding to "
+                                        + processor.getName()
+                                        + " does not exist."));
+    }
```

Comment:
As a final improvement here, I'd suggest making this a parameterized test instead of a single test with a loop. IMO, it's almost always preferable to do this when feasible as there's zero risk of for example there being no matching files (a parameterized test with empty parameterization throws an exception). In this case, I also think it increases the readability of the test.

A simple way to do it would be like so:

```suggestion
    @ParameterizedTest
    @MethodSource("processorFileProvider")
    public void test_eachProcessorIsAccompaniedByDescription(File processor) {
        assertTrue(
                getDescription(processor).isFile(),
                "Description corresponding to " + processor.getName() + " does not exist.");
    }

    private static Stream<Arguments> processorFileProvider() {
        return Processors.getAllProcessors().stream()
                .map(Class::getSimpleName)
                .map(procName -> PROCESSOR_PACKAGE.resolve(procName + ".java"))
                .map(Path::toFile)
                .map(Arguments::of);
    }
```

Reply:
> for example there being no matching files (a parameterized test with empty parameterization throws an exception)

Just to clarify: do you mean that if there are no processors, the test with method source would throw an error?

Reply:
Yes.

Reply:
When asserting in a loop (or stream) inside a test, you always run the risk of a vacuously passing test as the loop/stream is empty. That's why, when asserting in a loop, you should always first assert that collection you're iterating over is non-empty.